### ISSUE TITLE
XGBoost Regressor

### ISSUE THREAD

**Issue by Thilakraj1998** [blobcity/autoai#57]
Add XGBoostRegressor model into the library.

Primary File to Change: https://github.com/blobcity/autoai/blob/main/blobcity/config/regressor_config.py

Reference XGBoostRegressor Implementation:
https://github.com/blobcity/ai-seed/blob/main/Regression/XGBoost/XGBoostRegressor.ipynb

Also Refer offical documentation for required parameters:
https://xgboost.readthedocs.io/en/latest/python/python_api.html#module-xgboost.sklearn

Dependencies if any, must be appropriately added. Test run of the train function on a regression problem must pass, and the function must attempt to train a XGBoostRegressor as a potential best fit model.
